### DISCUSSION TITLE
Rate limiting

### DISCUSSION THREAD

**Discussion by Brotakuu** [supabase/supabase#4349] [Questions] [answered]
Is there any built-in rate limiting option for Auth (by ip)? 

Currently anyone can brute force with the endpoint and public key.

PS: We require username/password login so magic link will not work for us.

**Comment by kangmingtay** [comment, marked as answer]
Hey @Brotakuu, we're just in the midst of rolling this one out. We're also making the captcha configuration available on all projects soon so stay tuned!

**Reply by Sourav4479** [reply]
@kangmingtay Any update on this?

**Reply by 0xmax** [reply]
@kangmingtay Any update on this?

**Reply by dheereshag** [reply]
better go with appwrite

**Reply by kangmingtay** [reply]
we do have rate limits for this already - you can check out our rate limiting quotas here: https://supabase.com/docs/guides/platform/going-into-prod#auth-rate-limits

**Reply by ammarfaris** [reply]
Just curious, do you have rate-limiting per user, after user has logged-in ? say per user / or per IP address after user logged in to access certain end points to prevent user abuse? @kangmingtay 

This question is not inspired by twitter rate-limiting ruckus.

**Reply by kangmingtay** [reply]
@ammarfaris we don't provide a rate limit service for endpoints that are created by you

**Comment by naqibfaiyaz** [comment]
Hi am getting 429 for token endpoints, but the request count is not that high. Can you guys suggest anything for this?

**Reply by naqibfaiyaz** [reply]
I used the below query in the log_explorer to find out the request count and 429 count

```
select 
h.x_real_ip, 
Extract(minute from timestamp) as minutes, 
count(*) as request_count,
count(case when response.status_code=429 then 429 else null end) as count_429
  from edge_logs
  cross join unnest(metadata) as m
  cross join unnest(m.request) as request
  cross join unnest(m.response) as response
  LEFT JOIN UNNEST(m.request) AS r ON TRUE
  LEFT JOIN UNNEST(r.headers) AS h ON TRUE
  where (regexp_contains(event_message, '')) and (request.path like '/auth/%')
  and Extract(hour from timestamp)=9
  group by 1,2
  order by 2
```
  
  and the result is pretty insignificant request count
  
<img width="830" alt="image" src="https://user-images.githubusercontent.com/25023996/179781263-5bc05a12-7409-48cb-9506-9e5272ff4ab1.png">








There was only 1 request for a particular IP (163.47.158.234) which got 429

<img width="793" alt="image" src="https://user-images.githubusercontent.com/25023996/179781828-4cc64cde-d280-466b-95a3-1de8bd51de90.png">

**Comment by blarfoon** [comment]
Any update on this?

**Comment by kangmingtay** [comment]
Hey @killpowa and @Sourav4479, we have this already. You can refer to the rate limiting quotas here: https://supabase.com/docs/guides/platform/going-into-prod#rate-limits

**Reply by kangmingtay** [reply]
@0xmax

**Reply by FJohlinger** [reply]
I've looked at the link and I don't quite understand which of these rate limits apply to log-in attempts. Could anyone clarify this? Is this included in the 'Verification requests'?
